docs/rfcs/0002-storing-course-data.md

```diff
@@ -0,0 +1,135 @@
+## Storing course data
+
+### Abstract
+
+The aim of this RFC is to make decisions about the following:
+- The model definitions* that will be needed to store course data.
+- The plan for loading legacy course data, and keeping that data current
+  after we switch to using our own Open edX instance.
+
+Out of scope for this RFC:
+- Content management – we will just rely on Django admin for creating/updating
+  courses by hand.
+- Any kind of grading/certificate data.
+
+\* - _Much like MicroMasters, we will keep our own models for courses/programs
+  that will exist in parallel with the course data in edX_
+
+### Models
+
+Since this app has very similar needs, `Program`, `Course`, and `CourseRun` 
+models will very closely mirror the [models in MicroMasters](https://github.com/mitodl/micromasters/blob/master/courses/models.py), 
+minus a few fields that are not relevant for this app.
+
+`Program`
+- title
+- description
+- thumbnail (image)
+- readable_id (string - will be used exclusively for reporting purposes)
+- live (boolean - indicates whether the program is live in edX, et. al.)
+- source (string/enum - indicates the platform from which this was loaded/created. 
+  This will only be needed for record-keeping. See below for more detail.)
+
+`Course`
+- program (`Program` foreign key, **nullable** since a course doesn't necessarily belong to a program)
+- position_in_program (integer - indicates course sort order in a program)
+- title
+- description
+- thumbnail (image)
+- readable_id (string - will be used exclusively for reporting purposes)
+- live (same as above)
+- source (same as above)
+
+`CourseRun`
+- course (`Course` foreign key)
+- courseware_id (string - the ID for the course run as it exists in
+  the courseware platform, e.g.: an edX course key like `course-v1:MITxPRO+AMx+1T2019`)
+- courseware_url (string - the URL for the course on the courseware platform)
+- title
+- start_date
+- end_date
+- enrollment_start
+- enrollment_end
+- live (same as above)
+- platform (string/enum - indicates the courseware platform where this course
+  run exists)
+
+Some things to debate/pay particular attention to:
+
+1. Programs/courses/course runs offered across different platforms
+
+    1. `Program`s and `Course`s will each have a single record even if they exist
+      on multiple platforms. `source` will indicate from where the `Program` or 
+      `Course` was loaded/created. It will not indicate the platform on which they
+      are active.
+    1. `CourseRun.platform` - As opposed to the `source` field in `Program` and
+      `Course`, this field _will_ indicate the platform on which the course run
+      is/was active.
```

Comment:
@rhysyngsun Spoke with Ferdi and reevaluated this. With this approach we could have a single `Program`/`Course` that represents those things as they exist in all platforms. The switch between platforms would be represented in the `CourseRun.platform` field.